list/node.vue

```diff
@@ -0,0 +1,86 @@
+<script>
+import SortableTable from '@/components/SortableTable';
+import Poler from '@/utils/poler';
+import {
+  STATE, NAME, ROLES, VERSION, INTERNAL_EXTERNAL_IP, CPU, RAM
+} from '@/config/table-headers';
+
+import { METRIC } from '@/config/types';
+import { parseSi } from '@/utils/units';
+
+const METRICS_POLL_RATE_MS = 30000;
+const MAX_FAILURES = 2;
+
+export default {
+  name:       'ListNode',
+  components: { SortableTable },
+
+  props: {
+    schema: {
+      type:     Object,
+      required: true,
+    },
+
+    rows: {
+      type:     Array,
+      required: true,
+    },
+  },
+
+  data() {
+    const metricPoller = new Poler(this.loadMetrics, METRICS_POLL_RATE_MS, MAX_FAILURES);
+
+    metricPoller.start();
+
+    return {
+      headers:         [STATE, NAME, ROLES, VERSION, INTERNAL_EXTERNAL_IP, CPU, RAM],
+      metricPoller,
+      rowsWithMetrics: [],
+    };
+  },
+
+  beforeDestroy() {
+    this.metricPoller.stop();
+  },
+
+  methods: {
+    async loadMetrics() {
+      const schema = this.$store.getters['cluster/schemaFor'](METRIC.NODE);
+
+      if (schema) {
+        const rawMetrics = await this.$store.dispatch('cluster/findAll', { type: METRIC.NODE });
+        const clonedRows = await Promise.all(this.rows.map(row => this.$store.dispatch('cluster/clone', { resource: row })));
+        const metricsLookup = rawMetrics.reduce((agg, metric) => ({
+          ...agg,
+          [metric.id]: metric.usage
+        }), {});
+
+        this.rowsWithMetrics = clonedRows.map((row) => {
```

Comment:
This makes the table and rows shown in it no longer reactive to change events; They will be updated only when the poller fires every 30s.

There's a "private" object on all models which can store stuff like this directly in the original model (without it becoming part of the data that gets serialized if you save() it.) 
 https://github.com/rancher/dashboard/blob/master/models/rio.cattle.io.service.js#L200

Reply:
I ended up refactoring this to do the metrics lookup directly in the node model but relying on the page to update the metrics (could this be updated over the websocket?).

In scenarios like this I think it would be useful to have something a concept like ember-api-store's reference().

Reply:
Pretty sure there's some reason they can't be easily watched in k8s, but will ask.

I skipped references/hasMany for simplicity/because I didn't expect to need them, but should be possible to add.  I still want to go back & replace the proxies with actual classes for performance at some point so that would be a good time to do it.